src/Infrastructure/Persistence/ApplicationDbContext.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Codidact.Application.Common.Interfaces;
+using Codidact.Domain.Common;
+using Codidact.Domain.Entities;
+using Codidact.Domain.Extensions;
+using Microsoft.EntityFrameworkCore;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Codidact.Infrastructure.Persistence
+{
+    public class ApplicationDbContext : DbContext, IApplicationDbContext
+    {
+        public ApplicationDbContext(DbContextOptions options)
+            : base(options) { }
+
+        public DbSet<Member> Members { get; set; }
+        public DbSet<Community> Communities { get; set; }
+        public DbSet<TrustLevel> TrustLevels { get; set; }
+
+
+        public override Task<int> SaveChangesAsync(CancellationToken cancellationToken = new CancellationToken())
+        {
+            foreach (var entry in ChangeTracker.Entries<AuditableEntity>())
+            {
+                switch (entry.State)
+                {
+                    case EntityState.Added:
+                        // TODO: Modify for insert
+                        break;
+                    case EntityState.Modified:
+                        // TODO: Modify for update
+                        break;
+                }
+            }
+
+            return base.SaveChangesAsync(cancellationToken);
+        }
```

Comment:
Missing blank line after terminating brace

Reply:
Fixed in c5f6b8d5493c321063cd7a6d8b5631ae22289e25